src-tool/Pact/Analyze/Check.hs

```diff
@@ -444,9 +434,12 @@ verifyFunctionInvariants (CheckEnv tables _consts _pDefs moduleData caps gov _de
               pure $ case queryResult of
                  Left smtFailure -> Left $
                    CheckFailure info (SmtFailure smtFailure)
-                 Right pass      -> Right pass
-
-  where
+                 Right pass      -> Right (pass, warnings)
+        let
+          warnings = res ^.. folded . folded . folded . folded .folded
+          result = (fmap.fmap.fmap) fst res
```

Comment:
`folded.folded.folded.folded`? This is getting a little unfortunate... What is going on here, and why is this needed? What's the intent?

Reply:
The result of the invariant analysis is: `TableMap [Either CheckFailure (CheckSuccess, [VerificationWarning])]`. We need to extract the list of warnings `[VerificationWarning]` at this point.

Reply:
Ah, then all you need is:
```
res ^.. biplate
```

Reply:
`biplate` required a `Data` instance, which would populate through the entire pact code-base (`Info`, `Code`, `Failure` (from TC)) and FV (`TranslateFailreNoLoc`, ...).  Do we want this?

Reply:
I momentarily forgot about the `Data` requirement. No, we do not want that merely for this convenience.